Block Pattern: Contact - Centered link and social links

kishanjasani opened the pull request:
**Description**
Adding the Contact with centered link and social link block pattern.

Closes https://github.com/WordPress/twentytwentyfive/issues/60

**Screenshots**
https://prnt.sc/aBysxxXPeKcb

**Testing Instructions**

1. Apply the PR.
2. Add page or posts and Click on `Block Inserter`.
3. Go to pattern section and click on contact block pattern: https://prnt.sc/MjvfRD63R5GQ

huzaifaalmesbah commented:
![Huzaifa-20240821092227](https://github.com/user-attachments/assets/9a8ef267-f422-4a03-85f9-3346938db029)

rejaulalomkhan commented:
Dear @kishanjasani, Thank you for moving forward to resolve the issue.  
Based on Figma design PR design not matched. You should follow the Figma design. 
Heading should be two line and 2nd line have underlined.  
| Figma Design  | PR Design  |
| ------------- | ------------- |
| ![image](https://github.com/user-attachments/assets/ea40227e-95c8-4f6b-ad69-2c49b93de93b) | ![image](https://github.com/user-attachments/assets/d8b5fa4e-4720-4e60-91d3-6e949203341d)  |

kishanjasani commented:
> Thanks @kishanjasani for the PR. We need to make some changes. Please remove this code: `"iconColorValue":"#111111", `and also add the social icon URL `#`.

Thank you for the suggestions @huzaifaalmesbah. I have updated the markup as per you suggestions.

> Heading should be two line and 2nd line have underlined.

Thank you for the review @rejaulalomkhan. 
https://prnt.sc/7oll01CxojEs I have added a link to add the underline as paragraph or heading block do not have underline option.  Let me know your suggestions. To break it into the two line I have used `<br>`. Is it good idea?

carolinan commented on patterns/contact-centered-social-link.php:
```diff
@@ -0,0 +1,28 @@
+
+<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|80"},"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--80)">
+	<!-- wp:paragraph {"align":"center","style":{"elements":{"link":{"color":{"text":"var:preset|color|contrast"}}}},"textColor":"contrast","fontSize":"x-large"} -->
```
Contrast is the default text color, so it can be removed from the pattern. Link color has not been added to the theme yet, but can also be removed.

kishanjasani replied:
Thanks @carolinan I removed the text and link color.

carolinan commented on patterns/contact-centered-social-link.php:
```diff
@@ -0,0 +1,28 @@
+<!-- wp:group {"style":{"spacing":{"padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|80"},"blockGap":"var:preset|spacing|50"}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--80)">
+	<!-- wp:paragraph {"align":"center","fontSize":"x-large"} -->
+	<p class="has-text-align-center has-text-color has-link-color has-x-large-font-size"><?php echo wp_kses_post( _x( 'Got questions? <br><a href="#" rel="nofollow">Feel free to reach out.</a>', 'Heading of the Contact social link pattern', 'twentytwentyfive' ) ); ?></p>
```
Since the text and link color was removed, the class names must also be removed:
has-text-color has-link-color

kishanjasani replied:
Updated the code.
